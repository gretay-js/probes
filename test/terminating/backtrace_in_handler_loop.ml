let bt n =
  Printexc.get_callstack 9999
  |> Printexc.raw_backtrace_to_string
  (* Inlining depends on the optimization settings such as flambda *)
  |> Str.global_replace (Str.regexp_string " (inlined)") ""
  |> Printf.printf "%d:\n%s" n

let () =
  let rec loop n =
    if n = 0 then
      [%probe "backtrace" (bt 0)]
    else begin
      loop (n - 1);
      bt n
    end
  in
  loop 10

