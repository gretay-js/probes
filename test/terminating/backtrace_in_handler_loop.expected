0: is backtrace length > 10? true
1: is backtrace length > 10? true
2: is backtrace length > 10? true
3: is backtrace length > 10? false
4: is backtrace length > 10? false
5: is backtrace length > 10? false
6: is backtrace length > 10? false
7: is backtrace length > 10? false
8: is backtrace length > 10? false
9: is backtrace length > 10? false
10: is backtrace length > 10? false
